### ISSUE TITLE
Initial trackball line position

### ISSUE THREAD

**Issue by anisalibegic** [syncfusion/flutter-examples#59]
Is it possible to set initial trackball line position? I need a line to be visible when chart renders for the first time. Currently, it is visible only when touching the chart. Ideally, I would love to set the trackball line position using index.

**Comment by dharanidharandharmasivam** [comment]
Hi @anisalibegic ,

Greetings from Syncfusion. As of now, the trackball line is visible only on interaction with chart widget. However, we can achieve your requirement as a workaround, but we are having an issue with the workaround. The issue has been fixed and it is under the testing phase. The issue fix will be included in our upcoming volume 4 main release which is expected to be out in the middle of December 2019. We will notify you once the release is rolled out and we appreciate your patience until then.

Thanks,
Dharani.

**Comment by anisalibegic** [comment]
I saw you published **17.4.40** version. Does it include this feature?

**Comment by dharanidharandharmasivam** [comment]
Hi @anisalibegic ,

Yes, the issue has been resolved with the updated package and thanks for being patience. Now we have achieved your requirement as a workaround. Initially, we have rendered line series like trackball line and then on onTrackballPositionChanging event, we have hidden the line series. Find the code snippet to achieve this.

> SfCartesianChart(
          trackballBehavior: TrackballBehavior(
              enable: true, activationMode: ActivationMode.singleTap),
          onTrackballPositionChanging: (TrackballArgs args) {
            if (lineVisible == true)
              setState(() {
                lineVisible = false;
              });
          },
         series: <ChartSeries<TendenciasData, String>>[
            LineSeries<TendenciasData, String>(
                // Other configurations
            ),
            LineSeries<TendenciasData, String>(
                isVisible: lineVisible,
                // Other configurations 
                ),
          ]
)

Kindly use the below package to resolve the issue and find the sample below for the above workaround.

**Pub:** https://pub.dev/packages/syncfusion_flutter_charts/versions/17.4.40 
**Sample:** https://www.syncfusion.com/downloads/support/directtrac/256490/ze/chart_example-1730623755 

Thanks,
Dharani.

**Comment by anisalibegic** [comment]
> Hi @anisalibegic ,
> 
> Yes, the issue has been resolved with the updated package and thanks for being patience. Now we have achieved your requirement as a workaround. Initially, we have rendered line series like trackball line and then on onTrackballPositionChanging event, we have hidden the line series. Find the code snippet to achieve this.
> 
> > SfCartesianChart(
> > trackballBehavior: TrackballBehavior(
> > enable: true, activationMode: ActivationMode.singleTap),
> > onTrackballPositionChanging: (TrackballArgs args) {
> > if (lineVisible == true)
> > setState(() {
> > lineVisible = false;
> > });
> > },
> > series: <ChartSeries<TendenciasData, String>>[
> > LineSeries<TendenciasData, String>(
> > // Other configurations
> > ),
> > LineSeries<TendenciasData, String>(
> > isVisible: lineVisible,
> > // Other configurations
> > ),
> > ]
> > )
> 
> Kindly use the below package to resolve the issue and find the sample below for the above workaround.
> 
> **Pub:** https://pub.dev/packages/syncfusion_flutter_charts/versions/17.4.40
> **Sample:** https://www.syncfusion.com/downloads/support/directtrac/256490/ze/chart_example-1730623755
> 
> Thanks,
> Dharani.

_**The Syncfusion account you logged in to is not associated with this download. Please use the Syncfusion account associated with the file you are trying to download.**_

**Comment by dharanidharandharmasivam** [comment]
Hi @anisalibegic ,

Thanks for revert, but we are not clear about your query. So, we would like to know where you have received this error or provide more information on your query. This will be much helpful in providing the solution sooner. 

Thanks,
Dharani.

**Comment by anisalibegic** [comment]
I can't download the sample file from https://www.syncfusion.com/downloads/support/directtrac/256490/ze/chart_example-1730623755. It tells me that I'm not associated with the download.

**Comment by dharanidharandharmasivam** [comment]
Hi @anisalibegic ,

Thanks for your information. The provided link is working fine at our end. Even though we have created a new link for the sample which can be found from the below link. And if this also not worked at your end, kindly make sure that you have logged into the Syncfusion account.

https://www.syncfusion.com/downloads/support/directtrac/general/ze/chart_example-1730623755966504111 

Thanks,
Dharani.

**Comment by anisalibegic** [comment]
Hi @dharanidharandharmasivam!

If I want to call a ```trackball.show(x, y);``` I need to pass the X and Y coordinates. What if I don't have those coordinates? How to show a trackball line at specific index?

**Comment by dharanidharandharmasivam** [comment]
Hi @anisalibegic ,

As of now, you can show the trackball using the x and y coordinates only. However, we have considered your requirement as a feature to show the trackball based on the series/data point index and logged feature in our feedback portal. This will be available for volume 4 SP1 release which is expected to be rolled out at the end of January 2020. You can track the status of the feature request below.

https://www.syncfusion.com/feedback/10926 

Thanks,
Dharani.

**Comment by anisalibegic** [comment]
Hi @dharanidharandharmasivam,

Thank you very much. Is there any workaround to convert index to x & y coordinates until you release that version?

**Comment by dharanidharandharmasivam** [comment]
Hi @anisalibegic ,

Thanks for the revert. We have tried to achieve your requirement as a workaround, unfortunately, it’s not achievable as a workaround. So, as stated earlier in our previous updates we will provide your requirement as a feature. We appreciate your patience until then.

Thanks.
Dharani.

**Comment by anisalibegic** [comment]
Hi @dharanidharandharmasivam,

> [17.4.46] - 01/30/2020
> Features

> Provided support for displaying the trackball dynamically based on the data point index.

I saw you published a new version which includes the requested feature and thanks for that. I want to report a bug when changing trackball position by index. Everything works great when y values (sales in my project) are greater than 0. If at any point the y value is 0 then trackball is not visible and it will be visible when I move it the point that has the y value above 0. Is this by design?

**Comment by dharanidharandharmasivam** [comment]
Hi @anisalibegic ,

Yes, we have published the vol 4 sp 1 release (v14.4.46) with the requested feature. We have also fixed the reported issue with the below chart package. Now the trackball will be visible when the y-value is zero.

https://www.syncfusion.com/downloads/support/directtrac/general/ze/intermediate-198820266 

Download the chart widget package from above and refer the local package in your pubspec file. We will include the fix in our upcoming publish. Find the code snippet for referring the package locally,

```
syncfusion_flutter_charts:
    path: D:/flutter_charts/intermediate
```

Thanks,
Dharani.

**Comment by anisalibegic** [comment]
Hi @dharanidharandharmasivam,

Thanks for the fix. Is there a reason why the _pointIndex_ in the trackball show method starts at 1 and not the 0 like in the _onTrackballPositionChanging_ event?

When I say `_trackball.show(0, 0, 1);` it should show the trackball line at the second point from the left but it show it at the first point. Also, in _onTrackballPositionChanging_ it says that the first index is 0.

Why is that?

**Comment by dharanidharandharmasivam** [comment]
Hi @anisalibegic ,

We have analyzed the reported scenario. We would like to let you know that the show method of the trackball is having 3 arguments. The first two arguments specify the x and y positions (pixel) based on the trackball will be visible and the third argument is the pointIndex. 

If you wish to show the trackball based on the pointIndex, then specify the index to the third argument and the first two arguments is null. If both location (x and y values) and pointIndex is specified, then trackball will be visible based on pointIndex.

We have also ensured the reported scenario, unfortunately when pointIndex is specified as 1, then trackball is visible for the second point. Sample used for testing can be found below.

https://www.syncfusion.com/downloads/support/directtrac/general/ze/chart_tooltip-1626006483

Kindly try to replicate the reported scenario with the above sample or provide your sample. This will be helpful in further analysis and provide you the solution sooner.

Thanks,
Dharani
